### PR TITLE
PR for Reviewers to Comment

### PR DESCRIPTION
**Pull request by jessicl-ms**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

**Review by djj0809 — CHANGES_REQUESTED** [review]

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/afdx.json [hunk 1]
```diff
@@ -0,0 +1,6299 @@
...
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Cdn/validateSecret": {
```
[the remark this answers is not in the document]
Comment (jessicl-ms) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/afdx.json:3248]: DMed @huaiyizhu to ask, would Portal may want /ValidateSecret API for UrlSigningKey?

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/afdx.json [hunk 2]
```diff
@@ -0,0 +1,6299 @@
...
+      }
+    },
+    "KeyGroup": {
+      "description": "Contains a list of names of UrlSigningKey type secret objects contained in key group.",
+      "properties": {
+        "keys": {
+          "description": "Names of UrlSigningKey type secret objects",
+          "type": "array",
+          "items": {
+            "type": "string"
```
[the remark this answers is not in the document]
Comment (jessicl-ms) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/afdx.json]: Fixed

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/afdx.json [hunk 3]
```diff
@@ -0,0 +1,6299 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/KeyGroup"
+            }
+          },
+          "201": {
+            "description": "Created. The request has been fulfilled and a new delivery rule has been created.",
```
[the remark this answers is not in the document]
Comment (jessicl-ms) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/afdx.json]: Updated

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/examples/KeyGroups_Create.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        "keyReferences": [
+          {
+            "id":"/subscriptions/subid/resourcegroups/RG/providers/Microsoft.Cdn/profiles/profile1/secrets/secret1"
+          },
+          {
+            "id":"/subscriptions/subid/resourcegroups/RG/providers/Microsoft.Cdn/profiles/profile1/secrets/secret2"
+          },
+          {
+            "id":"/subscriptions/subid/resourcegroups/RG/providers/Microsoft.Cdn/profiles/profile1/secrets/secret3"
+          }
```
[the remark this answers is not in the document]
Comment (jessicl-ms) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/examples/KeyGroups_Create.json:19]: The PM spec says we support 1-5 keys per keygroup.

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/examples/Rules_Create_UrlSigningAction.json [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+                  "GET"
+                ],
+                "negateCondition": false,
+                "transforms": []
+              }
+            }
+          ],
+          "actions": [
+            {
+              "name": "ModifyResponseHeader",
```
[the remark this answers is not in the document]
Comment (jessicl-ms) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/examples/Rules_Create_UrlSigningAction.json]: Thanks for catching that. Fixed!

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/afdx.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jessicl-ms) [reply specification/cdn/resource-manager/Microsoft.Cdn/preview/2023-05-01-preview/afdx.json:3248]: Confirmed that we shuold support ValidateSecret API for UrlSigningKey

**Pull request closed without merging**